extension/parquet/parquet-extension.cpp

```diff
@@ -166,6 +166,15 @@ class ParquetScanFunction : public TableFunction {
 		return move(result);
 	}
 
+	static int ParquetProgress(ClientContext &context, const FunctionData *bind_data_p) {
+		auto &bind_data = (ParquetReadBindData &)*bind_data_p;
+		if (bind_data.initial_reader->NumRows() == 0) {
+			return 100;
+		}
+		auto percentage = bind_data.chunk_count * STANDARD_VECTOR_SIZE * 100 / bind_data.initial_reader->NumRows();
```

Comment:
This looks like it will only work for single-file parquet scans; could you check if there is a way of making it work with multi-file parquet scans as well?